From fc59f46bbf376af36f0c02c5f945a970f5d1eb0a Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <xiaowen.liu@freescale.com>
Date: Wed, 20 Jan 2016 10:07:31 -0500
Subject: [PATCH 629/635] MA-7514 Camera stops if take a high quality picture
 several times for C920.

The high quality resolution fps is slow for C920.
From the error log, dequeue buffer and compose one JPEG picture cost
long time for high quality resolution.
Filter low fps resolution for picture taken.

Signed-off-by: Xiaowen Liu <xiaowen.liu@freescale.com>
---
 mx6/libcamera3/UvcDevice.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mx6/libcamera3/UvcDevice.cpp b/mx6/libcamera3/UvcDevice.cpp
index ef3a9a6..807c317 100644
--- a/mx6/libcamera3/UvcDevice.cpp
+++ b/mx6/libcamera3/UvcDevice.cpp
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2012-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2016 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -124,8 +124,11 @@ status_t UvcDevice::initSensorStaticData()
         ALOGI("vid_frmval denominator:%d, numeraton:%d",
                 vid_frmval.discrete.denominator,
                 vid_frmval.discrete.numerator);
-        mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.width;
-        mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.height;
+        if (vid_frmval.discrete.denominator /
+                vid_frmval.discrete.numerator >= 5) {
+            mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.width;
+            mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.height;
+        }
 
         if (vid_frmval.discrete.denominator /
                 vid_frmval.discrete.numerator > 15) {
-- 
1.8.0

